framework/assets/yii.validation.js

```diff
@@ -244,7 +244,15 @@ yii.validation = (function ($) {
 
             value = $input.val();
             if (!options.skipOnEmpty || !pub.isEmpty(value)) {
-                value = $.trim(value);
+                value = new String(value);
+                if (options.chars || !String.prototype.trim) {
```

Comment:
I suggest to refactor it to use early return.

Reply:
@arogachev what's conditions to `early return`? can you send commit?

Reply:
Here is an article describing this principle https://medium.com/swlh/return-early-pattern-3d18a41bba8.

Reply:
@arogachev aaa... for me it's normal return)